docs(sca): Adding user flows. (#2467)
bonitasoft/bonita-doc: modules/build-run/pages/sca-user-flows.adoc

= Build and run an Application

> :page-aliases: ROOT:sca-user-flows.adoc

Reviewer: You can remove this alias, it's not necessary.

:experimental: // activate the 'menu' macro
:description: User flows possible in Application mode

== User flows in Application mode

=== Creating a new project in Studio using Application mode

[WARNING]
====
In order to generate the docker image, it's mandatory to have a maven repository working to get all the dependencies, including the latest Application runtime version.
====

> . Run Bonita Studio 2023.2 and create a new project.

Reviewer: ```suggestion
. Run Bonita Studio {bonitaVersion} and create a new project.
```

Reviewer: It's better to use Antora Attribute for this value. If you use 2023.2, we will need to update this value on next branch's version...

. Configure the project environment

. Validate the project
.. To work with Application mode, you need to explicitly mapping a form for each interaction form (instantiation and manual task)

. Admin and user app is now available in extension, if you want to deploy it on the final bundle with Application mode, you need to add it manually

.. Go to the Extension view

.. Open Marketplace and filter on “Application type”

.. You can now add the default application as you wish on the final bundle

. At the root of the project folder, run `mvn clean package -Pbundle,docker "-Dbonita.environment=production"` 

> . Find the docker image name with docker images to be sure it's being created.

Reviewer: ```suggestion
. Find the docker image name with `docker images` to be sure it's being created.
```

. Run the docker image with command (adapting licenses folder if needed) `docker run --rm --name MyProjectName -h MyProjectName -e -v ~/lic/:/opt/bonita_lic/ -d -p 9000:8080 MyProjectNameImage-production:0.0.1`

. Inspect logs with the command `docker logs` 

> . Go to `localhost:9000/bonita`, launch a process

Reviewer: ```suggestion
. Go to http://localhost:9000/bonita[http://localhost:9000/bonita], launch a process
```

. Your application is working! 

[NOTE]
====
In case you need to clean your docker: Stop and delete the docker container with docker rm -f MyProjectName and remove the docker image with docker rmi -f MyProjectNameImage-production:0.0.1
====

=== Migrate an existing project in Studio to Application mode

> . Run Bonita Studio 2023.2 and import from .bos or Git your project to migrate

Reviewer: ```suggestion
. Run Bonita Studio {bonitaVersion} and import from .bos or Git your project to migrate
```

. Migrate your project

. Validate your project

.. To work with Application mode, you need to explicitly mapping a form for each interaction form (instantiation and manual task)

. Admin and user app is now available in extension, if you want to deploy it on the final bundle with Application mode, you need to add it manually

.. Go to the Extension view

.. Open Marketplace and filter on “Application type”

.. You can now add the default application as you wish on the final bundle

. At the root of the project folder, run `mvn clean package -Pbundle,docker "-Dbonita.environment=qualification"`

. To run the docker images build in previous step, you need to run: `docker run --rm --name MyProjectName -h MyProjectName -v ~/lic/:/opt/bonita_lic/ -d -p 9000:8080 MyProjectNameImage-qualification:0.0.1`

>  .. If you need custom-permissions (in restApiExtensions as example), you can create another volume and map this volume with a local folder (with the right custom-mapping-permission file) with `-v ~/bonita-tenant-template-portal/:/opt/bonita/setup/platform_conf/initial/tenant_template_portal/`

Reviewer: Can you put a link to the docker options pages ?

> . Go to `localhost:9000/bonita`, launch a process

Reviewer: ```suggestion
. Go to http://localhost:9000/bonita[http://localhost:9000/bonita], launch a process
```
